service/module.go

```diff
@@ -48,57 +47,68 @@ type Module interface {
 
 // ModuleProviderFromFunc creates a new ModuleProvider from a name and create function.
 func ModuleProviderFromFunc(name string, createFunc func(Host) (Module, error)) ModuleProvider {
-	return &moduleProviderFunc{name, createFunc}
+	return &moduleProvider{name: name, createFunc: createFunc}
 }
 
 // ModuleOption is a function that configures module creation.
 type ModuleOption func(*moduleOptions) error
 
-// WithModuleName will override the name given by the ModuleProvider.
-func WithModuleName(name string) ModuleOption {
+// WithName will override the root service name specified in config.
+func WithName(name string) ModuleOption {
```

Comment:
names are confusing since we have WithModuleName. Can we make this "WithServiceName"?